File: 965-univalued-binary-tree/965-univalued-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSame(self,root,val):
        if not root: return True
        if(root.val!=val): return False
        return self.isSame(root.left,val) and self.isSame(root.right,val)
    
    def isUnivalTree(self, root: Optional[TreeNode]) -> bool:
        return self.isSame(root,root.val)

File: 965-univalued-binary-tree/965-univalued-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isSame(TreeNode* root, int val){
        if(!root) return true;
        if(root->val!=val) return false;
        return isSame(root->left,val) && isSame(root->right,val);
    }
    
    bool isUnivalTree(TreeNode* root) {
        return isSame(root,root->val);
    }
};